src/apps/Permissions/PermissionsView.js

```diff
@@ -97,25 +110,87 @@ function renderEntryExpansion({ entities, app, role }) {
       ))
 }
 
-function renderEntryActions(entry, onManageRole) {
-  return <EntryActions entry={entry} onManageRole={onManageRole} />
+function renderEntryActions(entry, onAssignPermission, onManageRole) {
+  return (
+    <EntryActions
+      entry={entry}
+      onAssignPermission={onAssignPermission}
+      onManageRole={onManageRole}
+    />
+  )
 }
 
 /* eslint-disable react/prop-types */
-function EntryActions({ entry, onManageRole }) {
-  const { role, app } = entry
+function EntryActions({ entry, onAssignPermission, onManageRole }) {
+  const theme = useTheme()
+  const { revokePermission } = usePermissions()
+
+  const { app, entities, manager, role } = entry
   const { proxyAddress } = app
+  const { address: entityAddress } = entities[0] || {}
   const roleBytes = role.bytes
 
-  const handleManageRoleClick = useCallback(() => {
+  const handleManageRole = useCallback(() => {
     onManageRole(proxyAddress, roleBytes)
   }, [roleBytes, proxyAddress, onManageRole])
 
+  const handleRevokePermission = useCallback(() => {
+    revokePermission({ appAddress: proxyAddress, entityAddress, roleBytes })
+  }, [revokePermission, entityAddress, proxyAddress, roleBytes])
+
+  const actions = []
+  if (isBurnEntity(manager)) {
+    actions.push([handleManageRole, IconView, 'View permission'])
+  } else {
+    if (isUnassignedEntity(manager)) {
+      actions.push([
+        handleManageRole,
+        IconEdit,
+        entities.length > 0 ? 'Re-initialize permission' : 'Create permission',
```

Comment:
@dizzypaty I've taken the liberty to make this a bit closer to their technical reality rather than "Set manager"

Reply:
Is it important for the user to know that'? Just wondering if they would know what initializing a permission means.

Reply:
Probably not, but "Set permission" is also inaccurate because this action also forces you to grant a permission.